src/sentry/hybridcloud/tasks/deliver_webhooks.py

```diff
@@ -222,6 +222,27 @@ def drain_mailbox_parallel(payload_id: int) -> None:
     request_failed = False
     delivered = 0
 
+    # Remove any payloads that have been backlogged for MAX_DELIVERY_AGE.
+    # Once payloads are this old they are low value, and we're better off prioritizing new work.
+    max_age = timezone.now() - MAX_DELIVERY_AGE
+    if payload.date_added < max_age:
+        deleted, _ = WebhookPayload.objects.filter(
+            id__gte=payload.id,
+            mailbox_name=payload.mailbox_name,
+            date_added__lte=timezone.now() - MAX_DELIVERY_AGE,
```

Comment:
Hmm do we need an index on date_added for this query? I wonder if we'll run into issues with scans across this column

Reply:
That is a good question, I'll pull up a query plan based on production data. I'm also a little concerned about the the first cleanup pass being heavy on the db.

Reply:
Switched to a delete using a subquery so that query costs are mostly constant.